### DISCUSSION TITLE
How do I run a script for multiple files/glob input?

### DISCUSSION THREAD

**Discussion by Gremious** [nushell/nushell#12611] [Q&A] [answered]
I have a folder with a few videos and I wanna occatinally run ffmpeg to extract their audio.

`./script.nu *.mp4` doesn't work, it says paramtere missing

`nu script.nu *.mp4` auto expands and now my funciton is expected to take in a specific number of string, not even a list or antying

I would pass in a glob string but 
a) it's annoyting to have to quote every time and 
b) The docs says that I'm supposed to use `into glob` to convert from string but that doesn't seem to work:

```nu
┌in ~\test
└❯ glob *.mp4
╭───┬──────────────────────────────────────────╮
│ 0 │ C:\Users\Gremious\test\angry_dog_265.mp4 │
│ 1 │ C:\Users\Gremious\test\who is nl.mp4     │
╰───┴──────────────────────────────────────────╯

............................[04:40:25 PM] [took 1ms]............................

┌in ~\test
└❯ '*.mp4' | into glob
*.mp4
```

Please, how do I just provide my script a glob/an arbitrary number of files and run `ffmpeg -i $input -vn -acodec copy $input` on all of them?

**Comment by ghost** [comment, marked as answer]
i think the thing you are looking for is `...`:

```nu
def main [...files] {
  print $files
}
```
->
```
$ ./test.nu *.txt
╭───┬───────────────────╮
│ 0 │ a.txt             │
│ 1 │ en_wos_nu.txt     │
│ 2 │ english.txt       │
│ 3 │ viewerlist.txt    │
╰───┴───────────────────╯
```

`glob` is a datatype like `path`, `date`, etc. `| into glob` changes a string into this filetype and does not expand it.

when passing aruments to a external command (or script), which does not have a definition nu expands globs (`*.txt` -> multiple arguments) in order to be backwards-compatible with POSIX (and generally easier to use). when defining a nushell function you have to define each argument and expanded globs are multiple arguments. therefore a single basic argument defined in nushell just accepts 1 argument (file). the `...` says "accept any amount and put it into a list".

**Reply by Gremious** [reply]
Thank you so much! Saviour
